src/server/game/AI/CoreAI/UnitAI.cpp

```diff
@@ -193,6 +193,35 @@ void UnitAI::DoCastAOE(uint32 spellId, bool triggered)
     me->CastSpell(nullptr, spellId, triggered);
 }
 
+BasicEvent* UnitAI::DoCastDelayed(Unit* victim, uint32 spellId, uint32 delay, bool triggered, bool aliveCheck)
+{
+    struct DelayedCast : public BasicEvent
+    {
+        DelayedCast(Unit *caster, Unit *target, uint32 spellId, bool triggered, bool aliveCheck) : BasicEvent(),
+            _caster(caster), _target(target), _spellId(spellId), _triggered(triggered), _aliveCheck(aliveCheck) { }
+
+        bool Execute(uint64, uint32)
+        {
+            if (_aliveCheck && _caster->IsAlive())
+                _caster->CastSpell(_target, _spellId, _triggered);
+            else
+                _caster->CastSpell(_target, _spellId, _triggered);
```

Comment:
nice copy paste from somewhere

Reply:
the idea from DoCastDelayed is not from me. Tell me the real author and I change the commit message

Reply:
Thats not what he is talking about - look closely at the code. If branch is exactly the same as else branch

Reply:
I don't see the problem. My trinity core branch is 3(?) years behind. So there might be some issues with different code

Reply:
Just look closely dude.. your condition doesn't make any sense

Reply:
You don't understand, I'm not talking about git branches but this https://en.wikipedia.org/wiki/Branch_(computer_science)

Reply:
ty: shauren